### PR TITLE
[stable/airflow] : support adding connections

### PR DESCRIPTION
**Pull request by antweiss**
@gsemet 
#### What this PR does / why we need it:

This adds support for providing Airflow connections as a part of the chart.
Without predefined connections most DAGs cannot execute.
Though in theory it is possible to define connections via environment variables  - it is very cumbersome, especially for connections with lots of 'extra' content (e.g Amazon EMR)

#### Special notes for your reviewer:

Currently this is implemented only for the scenario where init-container is enabled.
This allows extending the existing configmap-scripts template. 
Alternatively can be done in a separate configmap so as to decouple it from DAG loading logic.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/airflow/README.md
Comment (gsemet) [inline stable/airflow/README.md:113]: wonder if helm core team wont ask us to store these creds in a secret somehow.

**Comment by gsemet** [comment]
Thanks a lot for your contribution. That is indeed a missing block in this chart, storing airflow connections in the configuration.

/assign unguiculus

@unguiculus What's your opinion of my remark about credentials in the review?

File: stable/airflow/README.md
Comment (antweiss) [reply stable/airflow/README.md:113]: I agree. Trying to wrap my head around this as we speak. I would probably mean to put the whole of add-connections.sh in a secret and then mount and execute it. Instead of putting it in this configmap.
How does this sound?

File: stable/airflow/templates/configmap-scripts.yaml
Comment (bskim45) [inline stable/airflow/templates/configmap-scripts.yaml]: you've got wrong argument name 😅`--conn_extra`

**Review by bskim45 — CHANGES_REQUESTED** [review]

File: stable/airflow/templates/configmap-scripts.yaml
Comment (antweiss) [reply stable/airflow/templates/configmap-scripts.yaml]: Good catch! Thanks. Fixing ASAP.

**Comment by antweiss** [comment]
@unguiculus - I would love to get your opinion on this. Thanks!

**Comment by unguiculus** [comment]
@gsemet Credentials should go into secrets.

**Comment by antweiss** [comment]
Great, rewriting this to put connection properties into secrets.

**Comment by gsemet** [comment]
/lgtm

Wonderful !

**Comment by bskim45** [comment]
let's bump chart version to v1.12.0 (#10333)

**Comment by antweiss** [comment]
@bskim45 is it 1.12.0 or 0.12.0 ?

**Comment by gsemet** [comment]
seems to be another conflict

File: stable/airflow/templates/configmap-scripts.yaml
Comment (gsemet) [inline stable/airflow/templates/configmap-scripts.yaml]: can you keep the empty line?

File: stable/airflow/templates/secret-connections.yaml
Comment (gsemet) [inline stable/airflow/templates/secret-connections.yaml]: and add empty line here :)

File: stable/airflow/templates/configmap-scripts.yaml
Comment (antweiss) [reply stable/airflow/templates/configmap-scripts.yaml]: Yep, sorry about that.

File: stable/airflow/templates/secret-connections.yaml
Comment (antweiss) [reply stable/airflow/templates/secret-connections.yaml]: Done

**Comment by antweiss** [comment]
> seems to be another conflict

@gsemet is that about the conflict in Chart.yaml? Looks like it's something I need your help to resolve.

**Comment by gsemet** [comment]
not sure how to see the conflict, github won't show me... did you tried "git pull --rebase" ?

/lgtm

**Comment by antweiss** [comment]
Rebased on the latest master.

**Comment by antweiss** [comment]
@gsemet please take a look when you have time and lgtm if appropriate. Thanks!!

**Comment by gsemet** [comment]
Great :) look like I have the permission to approuve :)

I met you test the final version of the chart, I cannot do it right now. Do not hesitate to submit fixes if needed ! Thanks for your dev !

File: stable/airflow/values.yaml
Comment (ogirardot) [inline stable/airflow/values.yaml:94]: When trying to use this as documented, I get a `Warning: Merging destination map for chart 'airflow'. Cannot overwrite table item 'connections', with non table value: map[]` 
Am I doing this wrong or shouldn't this default value be a `[]` ?

Comment (antweiss) [reply stable/airflow/values.yaml:94]: @ogirardot that is weird. Works fine on my end with curly braces. Are you getting this on the initial install? Can you share your values.yaml?

File: stable/airflow/templates/deployments-scheduler.yaml
Comment (somcsel) [inline stable/airflow/templates/deployments-scheduler.yaml:108]: @antweiss Why is this included only in the block where `.Values.dags.initContainer.enabled` and `.Values.dags.initContainer.installRequirements` are true?

Comment (somcsel) [reply stable/airflow/templates/deployments-scheduler.yaml:108]: i.e. I think it should be included also in the else block on line 122: https://github.com/helm/charts/pull/10317/files#diff-f4f5de6f12364c16a36eaea2622e0361R122

Comment (somcsel) [reply stable/airflow/templates/deployments-scheduler.yaml:108]: PR to change the logic: #11495

File: stable/airflow/values.yaml
Comment (osterman) [reply stable/airflow/values.yaml:94]: It perhaps worked in earlier versions of helm (if it was less strict), but it's definitely not working on `2.14.1`. We're mixing datatypes here. As per the example above, it expects a list, yet the default value is being assigned as an empty map. As a result of this, it's not possible to pass any connections because of the error @ogirardot mentions.

Comment (osterman) [reply stable/airflow/values.yaml:94]: Also, for further clarification, it "works" with empty `{}` braces because the `Range` is empty. But as soon as one attempts to pass a non-empty `[]` list, then it fails.

Comment (gsemet) [reply stable/airflow/values.yaml:94]: Yes, this should be a list.

Comment (maximmi) [reply stable/airflow/values.yaml:94]: fixed in 4.0.1

**Pull request merged**
